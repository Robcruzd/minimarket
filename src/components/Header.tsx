import { Image, StyleSheet, TouchableOpacity, View } from "react-native";
import { Text } from "react-native";
import { useSelector } from "react-redux";

const Header = (props: { title: string; navigation:any}) => {

    const totalPrice = useSelector((state:any) => state.totalPriceSlice.totalPrice);

    const isShoppingCart = props.title === 'Shopping Cart';

    const handleScreen = () => {
        if(isShoppingCart){
            props.navigation.navigate("Home");
        }else {
            props.navigation.navigate("ShoppingCart")
        }
    }

    return (
        <View style={styles.viewHeader}>
            <View style={styles.viewArriba}>
                <Image
                    style={styles.imageLogo}
                    source={require("minimarket/src/images/logo.png")}
                />
                <TouchableOpacity 
                    testID="shoppingCartButton"
                    onPress={() => handleScreen()}
                    style={styles.touchTotal}>
                        <View style={[styles.viewPriceTotal, {backgroundColor: isShoppingCart ? "#ECBDF2" : "#A51BB6"}]}>
                            <Image
                                style={styles.imageCarrito}
                                source={isShoppingCart ? require("minimarket/src/images/shopping-cart-p.png") : require("minimarket/src/images/shopping-cart1.png")}
                            />
                            <Text style={[{color:isShoppingCart ? "#A51BB6":'white'}, styles.textTotalPrice]}>${totalPrice}</Text>
                        </View>
                    {isShoppingCart &&
                        <Text style={styles.textX}>X</Text>
                    }
                </TouchableOpacity>
            </View>
                <Text style={styles.textTitle}>{props.title}</Text>
        </View>
    )
}

export default Header

const styles = StyleSheet.create({
    viewHeader:{
        borderBottomWidth:1, 
        borderBottomColor: "#ECBDF2", 
        flexDirection: 'column', 
        margin: 5
    },
    viewArriba: { 
        flexDirection: 'row', 
        justifyContent:'space-between',
        alignItems:'center',
        margin: 5
    },
    imageLogo: {
        width:40, 
        height:40
    },
    touchTotal: {
        flexDirection:'row', 
        height:30
    },
    viewPriceTotal: { 
        flexDirection:'row', 
        height:30,
        padding:5
    },
    imageCarrito: {
        width:20, 
        height:20
    },
    textTotalPrice :{
        fontSize:16, 
        marginLeft:5
    },
    textX: {
        fontWeight:'bold',
        fontSize:12, 
        width:20, 
        height:30, 
        backgroundColor: "#A51BB6",
        color:"white",
        textAlign:'center',
        textAlignVertical: 'center'
    },
    textTitle: {
        color: "#A51BB6", 
        fontWeight:'bold', 
        fontSize:16
    }
})