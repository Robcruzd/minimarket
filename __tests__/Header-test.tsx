import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { useSelector } from 'react-redux';
import Header from '../src/components/Header';

jest.mock('react-redux', () => ({
  useSelector: jest.fn()
}));

describe('<Header />', () => {

  let navigation:any;

  beforeEach(() => {
    navigation = { navigate: jest.fn() };
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('renders correctly with a title and a logo', () => {
    (useSelector as jest.Mock).mockReturnValueOnce(0);
    const title = 'Home';
    const header = renderer.create(<Header title={title} navigation={navigation} />).toJSON();
    expect(header).toMatchSnapshot();
  });

  it('renders correctly with a title and a shopping cart', () => {
    (useSelector as jest.Mock).mockReturnValueOnce(10);
    const title = 'Shopping Cart';
    const header = renderer.create(<Header title={title} navigation={navigation} />).toJSON();
    expect(header).toMatchSnapshot();
  });

  it('navigates to Home when clicked on shopping cart on the Product or ShoppingCart page', () => {
    (useSelector as jest.Mock).mockReturnValueOnce(0);
    const title = 'Home';
    const header = renderer.create(<Header title={title} navigation={navigation} />);
    const shoppingCart = header.root.findByProps({ testID: 'shoppingCartButton' });
    act(() => shoppingCart.props.onPress());
    expect(navigation.navigate).toHaveBeenCalledWith('ShoppingCart');
  });

  it('navigates to ShoppingCart when clicked on the shopping cart on the Home page', () => {
    (useSelector as jest.Mock).mockReturnValueOnce(10);
    const title = 'Shopping Cart';
    const header = renderer.create(<Header title={title} navigation={navigation} />);
    const shoppingCart = header.root.findByProps({ testID: 'shoppingCartButton' });
    act(() => shoppingCart.props.onPress());
    expect(navigation.navigate).toHaveBeenCalledWith('Home');
  });

});
