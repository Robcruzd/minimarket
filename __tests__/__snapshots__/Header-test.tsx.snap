// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly with a title and a logo 1`] = `
<View
  style={
    {
      "borderBottomColor": "#ECBDF2",
      "borderBottomWidth": 1,
      "flexDirection": "column",
      "margin": 5,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 5,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/images/logo.png",
        }
      }
      style={
        {
          "height": 40,
          "width": 40,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "height": 30,
          "opacity": 1,
        }
      }
      testID="shoppingCartButton"
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
              "height": 30,
              "padding": 5,
            },
            {
              "backgroundColor": "#A51BB6",
            },
          ]
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/images/shopping-cart1.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
        <Text
          style={
            [
              {
                "color": "white",
              },
              {
                "fontSize": 16,
                "marginLeft": 5,
              },
            ]
          }
        >
          $
          0
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#A51BB6",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Home
  </Text>
</View>
`;

exports[`<Header /> renders correctly with a title and a shopping cart 1`] = `
<View
  style={
    {
      "borderBottomColor": "#ECBDF2",
      "borderBottomWidth": 1,
      "flexDirection": "column",
      "margin": 5,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 5,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/images/logo.png",
        }
      }
      style={
        {
          "height": 40,
          "width": 40,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "height": 30,
          "opacity": 1,
        }
      }
      testID="shoppingCartButton"
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
              "height": 30,
              "padding": 5,
            },
            {
              "backgroundColor": "#ECBDF2",
            },
          ]
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/images/shopping-cart-p.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
        <Text
          style={
            [
              {
                "color": "#A51BB6",
              },
              {
                "fontSize": 16,
                "marginLeft": 5,
              },
            ]
          }
        >
          $
          10
        </Text>
      </View>
      <Text
        style={
          {
            "backgroundColor": "#A51BB6",
            "color": "white",
            "fontSize": 12,
            "fontWeight": "bold",
            "height": 30,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 20,
          }
        }
      >
        X
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#A51BB6",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Shopping Cart
  </Text>
</View>
`;
